### PR TITLE
Added attack animation.

### PR DESCRIPTION
**Pull request by ScottXileChen**
Added attack animation.
Update PlayerMovement, MariaAC scripts.
Added Targetable, InputTarget, and HeroCombat scripts.

### CODE REVIEW

File: WarSpecialists/Assets/Player/Scripts/HeroCombat.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public HeroAttackType heroAttackType;
+
+    public GameObject targetedEnemy;
+    public float attackRange;
+    public float rotateSpeedForAttack;
+
+    private PlayerMovement movement;
+
+    public bool basicAtkIdle = false;
+    public bool isHeroAlive;
```
Comment (erickbordallo) [inline WarSpecialists/Assets/Player/Scripts/HeroCombat.cs:18]: Default initialize this variable with `true`.

File: WarSpecialists/Assets/Player/Scripts/HeroCombat.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+using UnityEngine.AI;
+
+[RequireComponent(typeof(PlayerMovement))]
+public class HeroCombat : MonoBehaviour
+{
+    public enum HeroAttackType { Melee, Ranged }
+    public HeroAttackType heroAttackType;
+
+    public GameObject targetedEnemy;
+    public float attackRange;
```
Comment (erickbordallo) [inline WarSpecialists/Assets/Player/Scripts/HeroCombat.cs]: I think this attack range should come from a base class.

File: WarSpecialists/Assets/Player/Scripts/HeroCombat.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                        attackTimer -= Time.deltaTime;
+                        if (attackTimer < 0f)
+                        {
+                            attackTimer = attackDelayTime;
+                            // EnemyTakeDamage here
+                            Debug.Log("Attack the minion");
+                        }
+                    }
+                }
+
```
Comment (erickbordallo) [inline WarSpecialists/Assets/Player/Scripts/HeroCombat.cs]: Remove this extra line

File: WarSpecialists/Assets/Player/Scripts/HeroCombat.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            }
+
```
Comment (erickbordallo) [inline WarSpecialists/Assets/Player/Scripts/HeroCombat.cs]: remove this extra line

File: WarSpecialists/Assets/Player/Scripts/EnemySide/Targetable.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Targetable : MonoBehaviour
+{
+    public enum EnemyType { Minion }
```
Comment (erickbordallo) [inline WarSpecialists/Assets/Player/Scripts/EnemySide/Targetable.cs]: Increase this enum list with: 
Tower, Champion and Base.

**Review by erickbordallo — APPROVED** [review]
Some small changes. Apply this changes and then you can merge. Nice work!

**Review by ScottXileChen — COMMENTED** [review]
Put value of attackRange to playerbase class

**Review by erickbordallo — APPROVED** [review]

**Pull request merged**